tests/test_hacks.py

```diff
@@ -0,0 +1,25 @@
+"""Tests of the --travis-after flag."""
+from tox_travis.hacks import subcommand_test_monkeypatch
+
+
+class TestSessionSubcommandTest:
+    """Test the overridden post hook for subcommand_test."""
+
+    def test_subcommand_test_post_hook(self, mocker):
+        """It should return the same retcode, but run the hook."""
+        subcommand_test = mocker.patch('tox.session.Session.subcommand_test',
+                                       return_value=42)
+        tox_subcommand_test_post = mocker.Mock()
+        subcommand_test_monkeypatch(tox_subcommand_test_post)
+
+        import tox.session
+        session = mocker.Mock()
+
+        real_subcommand_test = tox.session.Session.subcommand_test
+        # Python 2 compat
+        real_subcommand_test = getattr(
+            real_subcommand_test, '__func__', real_subcommand_test)
```

Comment:
Out of curiosity, is this necessary due to tox using old-style classes?

Reply:
Yep.

Reply:
Actually, no. In Python 2 the function's descriptor makes an unbound `instancemethod`, which enforces that the first argument must be of the type that the method was created on. In Python 3 an unbound method, accessed directly on the class, is just directly accessing the function. In this case I didn't want to create a real `Session` object (even mocking with a Spec didn't work because the `config` property is added dynamically), so I had to unwrap the actual function so I could call it.

```pycon
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Spam(object):
...  def x(self):
...   print('spam')
... 
>>> Spam.x(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unbound method x() must be called with Spam instance as first argument (got NoneType instance instead)
>>> type(Spam.x)
<type 'instancemethod'>
>>> type(Spam().x)
<type 'instancemethod'>
```

```pycon
Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class Spam(object):
...  def x(self):
...   print('spam')
... 
>>> Spam.x(None)
spam
>>> type(Spam.x)
<class 'function'>
>>> type(Spam().x)
<class 'method'>
```